assets/terraform-modules/packet/flatcar-linux/kubernetes/workers/cl/worker.yaml.tmpl

```diff
@@ -421,6 +431,41 @@ storage:
           done
           echo "$record.$zone is available on all nameservers"
           exit 0
+    - path: /opt/bgp-metadata
+      filesystem: root
+      mode: 0544
+      contents:
+        inline: |
+          #!/bin/bash
+          set -o pipefail
+          max_attempts=3600
+          target=/run/metadata/bgp
+          echo "Polling metadata service for BGP information"
+          counter=0
+          while [[ $counter -lt $max_attempts ]]; do
+              out=$(curl -s -f --connect-timeout 5 \
+                  https://metadata.packet.net/metadata | jq -r .bgp_neighbors[0].peer_ips[0])
+              ret=$?
+              if [[ $ret -ne 0 ]]; then
+                  echo "Non-zero exit code: $ret"
+              elif [[ "$out" = "" ]]; then
+                  echo "Empty response"
+              elif [[ "$out" = "null" ]]; then
+                  echo "Null response"
+              else
+                  echo "BGP metadata is available!"
+                  if [[ ! "$out" =~ ^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
```

Comment:
Hmm, maybe https://stackoverflow.com/a/5284410/2974814 ? :smile:

Reply:
Not sure what you're suggesting. Could you explain?

Reply:
If you insist on validating obtained data, perhaps we should do it right. So using `^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\.|$)){4}$`, as current regex allows values like:
```
256.0.0.0
1.1.1.1.1
```

Which are not correct.

I was laughing, because validating IP addresses with regex is always fun :)

Reply:
I don't "insist" on anything, just thought it's better to add at least a minimal validation to catch at least cases where the output is obviously not an IP address. This isn't comprehensive and doesn't handle IPv6 but it's better than nothing.

`1.1.1.1.1` doesn't match on my testing. `256.0.0.0` indeed matches, but so do `0.0.0.0` and `127.0.0.1` which are valid IPs which don't make sense as a peer address...

I'm happy to use a tighter regex if we have one which works reliably.

Reply:
I've replaced the regex with `^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\.|$)){4}$` and am testing.